mhz19.cpp: unused function warning detection_range_to_log_string in 2026.3.0

### The problem

When compiling with ESPHome 2026.3.0, the following compiler warning appears:
src/esphome/components/mhz19/mhz19.cpp:19:25: warning: 'const esphome::LogString* esphome::mhz19::detection_range_to_log_string(esphome::mhz19::MHZ19DetectionRange)' defined but not used [-Wunused-function]
The function detection_range_to_log_string() is defined in mhz19.cpp but never called. This appears to be an oversight — likely a function prepared for future use or left over after a refactor.
Expected behaviour: No compiler warnings on a clean build.
ESPHome version: 2026.3.0
Board: ESP8266 NodeMCU v2

### Which version of ESPHome has the issue?

2026.3.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP8266

### Component causing the issue

_No response_

### YAML Config

```yaml

```

### Anything in the logs that might be useful for us?

```txt

```

### Additional information

_No response_

--- a/esphome/components/mhz19/mhz19.cpp
+++ b/esphome/components/mhz19/mhz19.cpp
@@ -16,6 +16,7 @@ static const uint8_t MHZ19_COMMAND_DETECTION_RANGE_0_2000PPM[] = {0xFF, 0x01, 0x
 static const uint8_t MHZ19_COMMAND_DETECTION_RANGE_0_5000PPM[] = {0xFF, 0x01, 0x99, 0x00, 0x00, 0x00, 0x13, 0x88};
 static const uint8_t MHZ19_COMMAND_DETECTION_RANGE_0_10000PPM[] = {0xFF, 0x01, 0x99, 0x00, 0x00, 0x00, 0x27, 0x10};
 
+#if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_DEBUG
 static const LogString *detection_range_to_log_string(MHZ19DetectionRange range) {
   switch (range) {
     case MHZ19_DETECTION_RANGE_0_2000PPM:
@@ -28,6 +29,7 @@ static const LogString *detection_range_to_log_string(MHZ19DetectionRange range)
       return LOG_STR("default");
   }
 }
+#endif
 
 uint8_t mhz19_checksum(const uint8_t *command) {
   uint8_t sum = 0;

